File: ABC/abc018/abc018_3 1.py
#30点解法
R, C, K = map(int, input().split())
s = [input() for _ in range(R)]

ans = 0
for x in range(K-1, R-K+1):
    for y in range(K-1, C-K+1):
        flag = True
        for i in range(R):
            for j in range(C):
                if abs(i-x) + abs(j-y) <= K-1 and s[i][j] == 'x':
                    flag = False
        if flag:
            ans += 1

print(ans)

File: ABC/abc018/abc018_3 1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // 30点解法
    int R, C, K, ans = 0;
    cin >> R >> C >> K;

    vector<string> s(R);
    for (int i = 0; i < R; i++) cin >> s[i];

    for (int x = K-1; x <= R-K; x++) {
        for (int y = K-1; y <= C-K; y++) {
            bool flag = true;
            for (int i = 0; i < R; i++) {
                for (int j = 0; j < C; j++) {
                    if (abs(i-x) + abs(j-y) <= K-1 && s[i][j] == 'x') {
                        flag = false;
                    }
                }
            }

            if (flag) ans++;
        }
    }

    cout << ans << endl;
}